core/src/main/java/org/polypheny/db/catalog/impl/logical/RelationalCatalog.java

```diff
@@ -0,0 +1,595 @@
+    public LogicalColumn addColumn( String name, long tableId, int position, PolyType type, PolyType collectionsType, Integer length, Integer scale, Integer dimension, Integer cardinality, boolean nullable, Collation collation ) {
+        long id = idBuilder.getNewFieldId();
+        LogicalColumn column = new LogicalColumn( id, name, tableId, logicalNamespace.id, position, type, collectionsType, length, scale, dimension, cardinality, nullable, collation, null );
+        columns.put( id, column );
+        change();
+        return column;
+    }
+
+
+    @Override
+    public void renameColumn( long columnId, String name ) {
+        columns.put( columnId, columns.get( columnId ).toBuilder().name( name ).build() );
+        change();
+    }
+
+
+    @Override
+    public void setColumnPosition( long columnId, int position ) {
+        columns.put( columnId, columns.get( columnId ).toBuilder().position( position ).build() );
+        change();
+    }
+
+
+    @Override
+    public void setColumnType( long columnId, PolyType type, PolyType collectionsType, Integer length, Integer scale, Integer dimension, Integer cardinality ) {
+        if ( scale != null && scale > length ) {
+            throw new RuntimeException( "Invalid scale! Scale can not be larger than length." );
+        }
+
+        columns.put( columnId, columns.get( columnId ).toBuilder().type( type ).length( length ).scale( scale ).dimension( dimension ).cardinality( cardinality ).build() );
+        change();
+    }
+
+
+    @Override
+    public void setNullable( long columnId, boolean nullable ) {
+        columns.put( columnId, columns.get( columnId ).toBuilder().nullable( nullable ).build() );
+        change();
+    }
+
+
+    @Override
+    public void setCollation( long columnId, Collation collation ) {
+        columns.put( columnId, columns.get( columnId ).toBuilder().collation( collation ).build() );
+        change();
+    }
+
+
+    @Override
+    public void deleteColumn( long columnId ) {
+        columns.remove( columnId );
+        change();
+    }
+
+
+    @Override
+    public LogicalColumn setDefaultValue( long columnId, PolyType type, PolyValue defaultValue ) {
+        LogicalColumn column = columns.get( columnId ).toBuilder().defaultValue( new LogicalDefaultValue( columnId, type, defaultValue, "defaultValue" ) ).build();
+        columns.put( columnId, column );
+        change();
+        return column;
+    }
+
+
+    @Override
+    public void deleteDefaultValue( long columnId ) {
+        columns.put( columnId, columns.get( columnId ).toBuilder().defaultValue( null ).build() );
+        change();
+    }
+
+
+    @Override
+    public LogicalTable addPrimaryKey( long tableId, List<Long> columnIds ) {
+        if ( columnIds.stream().anyMatch( id -> columns.get( id ).nullable ) ) {
+            throw new GenericRuntimeException( "Primary key is not allowed to use nullable columns." );
+        }
+
+        // TODO: Check if the current values are unique
```

Comment:
Todo

Reply:
I would leave that for now, as this was never implemented and we have to handle index in the future anyway.

Reply:
ok